src/main/java/fi/dy/masa/servux/dataproviders/StructureDataProvider.java

```diff
@@ -106,17 +168,30 @@ public void onStartedWatchingChunk(ServerPlayerEntity player, WorldChunk chunk)
 
     public boolean register(ServerPlayerEntity player)
     {
-        // System.out.printf("register\n");
         boolean registered = false;
+        MinecraftServer server = player.getServer();
         UUID uuid = player.getUuid();
 
         if (this.registeredPlayers.containsKey(uuid) == false)
         {
-            PacketSplitter.sendPacketTypeAndCompound(StructureDataPacketHandler.CHANNEL, StructureDataPacketHandler.PACKET_S2C_METADATA, this.metadata, player);
-
             this.registeredPlayers.put(uuid, new PlayerDimensionPosition(player));
-            int tickCounter = player.getServer().getTicks();
-            this.initialSyncStructuresToPlayerWithinRange(player, player.getServer().getPlayerManager().getViewDistance(), tickCounter);
+            int tickCounter = server.getTicks();
+
+            //Servux.logger.info("registering structures for player {}", player.getName().getLiteralString());
+
+            ServerPlayNetworkHandler handler = player.networkHandler;
+
+            if (handler != null)
+            {
+                NbtCompound nbt = new NbtCompound();
+                nbt.copyFrom(this.metadata);
+                nbt.putInt("packetType", ServuxStructuresHandler.PACKET_S2C_METADATA);
+
+                // Using the networkHandler method allows this to work
+                HANDLER.sendPlayPayload(new ServuxStructuresPayload(nbt), handler);
```

Comment:
uses networkHandler to send the initial Metadata, because Fabric API might not work this quickly.